/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package audio_focus_control_proto;

import "google/protobuf/empty.proto";

message AudioFocusRequest {
    uint64 session_id = 1;
    int32 audio_usage = 2;
    int32 zone_id = 3;
    bool allow_duck = 4;
    bool is_transient = 5;
    bool is_exclusive = 6;
}

message AudioFocusControlMessage {
    repeated AudioFocusRequest acquire_requests = 1;
    repeated uint64 release_requests = 2;
    repeated uint64 active_sessions = 3;
}

service AudioFocusControlServer {
    rpc AudioRequests(AudioFocusControlMessage) returns (google.protobuf.Empty) {}
}
